File: src/cotakwira/entities/Rate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cotakwira.entities;

import java.sql.Date;
import org.controlsfx.control.Rating;

/**
 *
 * @author yassin
 */
public class Rate {

    private int idRate, idCoach;
    Date DateRate;
    private int Rate;
    private String NomPrenomCoach, NomEquipe;
    Rating RateView;

    public Rate() {
    }

    public Rate(int idRate, int idCoach, int Rate, String NomEquipe) {
        this.idRate = idRate;
        this.idCoach = idCoach;
        this.Rate = Rate;
        this.NomEquipe = NomEquipe;
    }

    public Rate(int idRate, int idCoach, Rating RateView, String NomEquipe) {
        this.idRate = idRate;
        this.idCoach = idCoach;
        this.RateView = RateView;
        this.NomEquipe = NomEquipe;
    }

    public Rating getRateView() {
        return RateView;
    }

    public void setRateView(Rating RateView) {
        this.RateView = RateView;
    }

    
    public Rate(int idRate) {
        this.idRate = idRate;
    }

    public int getIdRate() {
        return idRate;
    }

    public void setIdRate(int idRate) {
        this.idRate = idRate;
    }

    public int getIdCoach() {
        return idCoach;
    }

    public void setIdCoach(int idCoach) {
        this.idCoach = idCoach;
    }

    public Date getDateRate() {
        return DateRate;
    }

    public void setDateRate(Date DateRate) {
        this.DateRate = DateRate;
    }

    public int getRate() {
        return Rate;
    }

    public void setRate(int Rate) {
        this.Rate = Rate;
    }

    public String getNomPrenomCoach() {
        return NomPrenomCoach;
    }

    public void setNomPrenomCoach(String NomPrenomCoach) {
        this.NomPrenomCoach = NomPrenomCoach;
    }

    public String getNomEquipe() {
        return NomEquipe;
    }

    public void setNomEquipe(String NomEquipe) {
        this.NomEquipe = NomEquipe;
    }

    @Override
    public String toString() {
        return "Rate{" + "idRate=" + idRate + ", idCoach=" + idCoach + ", DateRate=" + DateRate + ", Rate=" + Rate + ", NomPrenomCoach=" + NomPrenomCoach + ", NomEquipe=" + NomEquipe + '}';
    }

}

File: Rate.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;


/**
 * Rate
 *
 * @ORM\Table(name="rate", indexes={@ORM\Index(name="idCoach", columns={"idCoach"})})
 * @ORM\Entity
 */
class Rate
{
    /**
     * @var int
     *
     * @ORM\Column(name="idRate", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idrate;

    /**
     * @var int|null
     *
     * @ORM\Column(name="idCoach", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $idcoach = NULL;

    /**
     * @var string|null
     *
     * @ORM\Column(name="NomPrenomCoach", type="string", length=50, nullable=true, options={"default"="NULL"})
     */
    private $nomprenomcoach = 'NULL';

    /**
     * @var string|null
     *
     * @ORM\Column(name="NomEquipe", type="string", length=50, nullable=true, options={"default"="NULL"})
     */
    private $nomequipe = 'NULL';

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="DateRate", type="date", nullable=true, options={"default"="NULL"})
     */
    private $daterate = 'NULL';

    /**
     * @var int|null
     *
     * @ORM\Column(name="Rate", type="integer", nullable=true, options={"default"="NULL"})
     */
    private $rate = NULL;

    public function getIdrate(): ?int
    {
        return $this->idrate;
    }

    public function getIdcoach(): ?int
    {
        return $this->idcoach;
    }

    public function setIdcoach(?int $idcoach): self
    {
        $this->idcoach = $idcoach;

        return $this;
    }

    public function getNomprenomcoach(): ?string
    {
        return $this->nomprenomcoach;
    }

    public function setNomprenomcoach(?string $nomprenomcoach): self
    {
        $this->nomprenomcoach = $nomprenomcoach;

        return $this;
    }

    public function getNomequipe(): ?string
    {
        return $this->nomequipe;
    }

    public function setNomequipe(?string $nomequipe): self
    {
        $this->nomequipe = $nomequipe;

        return $this;
    }

    public function getDaterate(): ?\DateTimeInterface
    {
        return $this->daterate;
    }

    public function setDaterate(?\DateTimeInterface $daterate): self
    {
        $this->daterate = $daterate;

        return $this;
    }

    public function getRate(): ?int
    {
        return $this->rate;
    }

    public function setRate(?int $rate): self
    {
        $this->rate = $rate;

        return $this;
    }



}
